### DISCUSSION TITLE
v2.6.0

### DISCUSSION THREAD

**Discussion by timmo001** [timmo001/system-bridge#853] [General]
## ✨ New features

- Add scheduler and start and stop ws handler @timmo001 (#721)
- CLI flags and Docker Container @timmo001 (#751)
- Frontend framework rewrite using Next.js @timmo001 (#769)

## 🐛 Bug fixes

- Reduce number of open files @timmo001 (#749)
- Process manager @timmo001 (#825)
- Fix Windows Sensors @timmo001 (#852)

## 🚀 Enhancements

- Run observer tasks in worker @timmo001 (#713)
- Container use package @timmo001 (#763)
- Disable some settings when in container  @timmo001 (#786)
- Add yarn autoclean @timmo001 (#788)
- Process manager @timmo001 (#825)

## 🧰 Maintenance

- Restructure @timmo001 (#740)
- Logger class @timmo001 (#741)
- Container use package @timmo001 (#763)

## 📚 Documentation

- Add Android companion app links @timmo001 (#826)

## ⬆️ Dependency updates

- Bump concurrently from 6.2.0 to 6.2.1 @dependabot (#694)
- Bump @typescript-eslint/eslint-plugin from 4.29.0 to 4.29.1 @dependabot (#693)
- Bump ts-node from 10.1.0 to 10.2.0 @dependabot (#692)
- Bump @typescript-eslint/parser from 4.29.0 to 4.29.1 @dependabot (#691)
- Bump @types/jest from 26.0.24 to 27.0.0 @dependabot (#696)
- Bump supertest from 6.1.4 to 6.1.5 @dependabot (#695)
- Bump @types/react from 17.0.16 to 17.0.17 in /frontend @dependabot (#700)
- Bump @types/jest from 26.0.24 to 27.0.0 in /frontend @dependabot (#699)
- Bump @types/node from 16.4.13 to 16.4.14 in /frontend @dependabot (#698)
- Bump @types/node from 16.4.13 to 16.4.14 @dependabot (#697)
- Bump ws from 7.5.3 to 8.1.0 @dependabot (#702)
- Bump url-parse from 1.5.1 to 1.5.3 in /frontend @dependabot (#701)
- Bump @types/node from 16.4.14 to 16.6.0 in /frontend @dependabot (#704)
- Bump @types/node from 16.4.14 to 16.6.0 @dependabot (#703)
- Bump @types/jest from 27.0.0 to 27.0.1 @dependabot (#708)
- Bump typeorm from 0.2.36 to 0.2.37 @dependabot (#707)
- Bump @types/node from 16.6.0 to 16.6.1 @dependabot (#709)
- Bump @types/node from 16.6.0 to 16.6.1 in /frontend @dependabot (#710)
- Bump @types/jest from 27.0.0 to 27.0.1 in /frontend @dependabot (#711)
- Bump @types/react from 17.0.17 to 17.0.18 in /frontend @dependabot (#725)
- Bump @typescript-eslint/parser from 4.29.1 to 4.29.2 @dependabot (#724)
- Bump ts-jest from 27.0.4 to 27.0.5 @dependabot (#730)
- Bump system-bridge-windows-sensors from 1.2.1 to 1.2.2 @dependabot (#729)
- Bump dessant/lock-threads from 2.1.1 to 2.1.2 @dependabot (#728)
- Bump @typescript-eslint/eslint-plugin from 4.29.1 to 4.29.2 @dependabot (#723)
- Bump ibiqlik/action-yamllint from 3.0.2 to 3.0.3 @dependabot (#731)
- Bump supertest from 6.1.5 to 6.1.6 @dependabot (#732)
- Bump ts-node from 10.2.0 to 10.2.1 @dependabot (#733)
- Bump @types/react from 17.0.18 to 17.0.19 in /frontend @dependabot (#738)
- Bump ibiqlik/action-yamllint from 3.0.3 to 3.0.4 @dependabot (#734)
- Bump @types/node from 16.6.1 to 16.6.2 @dependabot (#736)
- Bump @types/node from 16.6.1 to 16.6.2 in /frontend @dependabot (#737)
- Bump eslint-plugin-prettier from 3.4.0 to 3.4.1 @dependabot (#743)
- Bump @types/node from 16.6.2 to 16.7.1 @dependabot (#742)
- Bump @typescript-eslint/parser from 4.29.2 to 4.29.3 @dependabot (#744)
- Bump @types/node from 16.6.2 to 16.7.1 in /frontend @dependabot (#747)
- Bump @typescript-eslint/eslint-plugin from 4.29.2 to 4.29.3 @dependabot (#746)
- Bump @nestjs/schematics from 8.0.2 to 8.0.3 @dependabot (#745)
- Bump systeminformation from 5.8.0 to 5.8.2 @dependabot (#748)
- Bump tsconfig-paths from 3.10.1 to 3.11.0 @dependabot (#750)
- Bump @types/node from 16.7.1 to 16.7.2 in /frontend @dependabot (#755)
- Bump systeminformation from 5.8.2 to 5.8.6 @dependabot (#754)
- Bump @types/node from 16.7.1 to 16.7.2 @dependabot (#753)
- Bump typescript from 4.3.5 to 4.4.2 @dependabot (#756)
- Bump jest from 27.0.6 to 27.1.0 @dependabot (#757)
- Bump @types/node from 16.7.2 to 16.7.3 @dependabot (#758)
- Bump @types/node from 16.7.2 to 16.7.4 in /frontend @dependabot (#762)
- Bump @types/node from 16.7.2 to 16.7.4 @dependabot (#761)
- Bump @types/node from 16.7.4 to 16.7.9 in /frontend @dependabot (#778)
- Bump eslint-plugin-prettier from 3.4.1 to 4.0.0 @dependabot (#777)
- Bump @types/node from 16.7.4 to 16.7.9 @dependabot (#776)
- Bump react-router-dom from 5.2.0 to 5.2.1 in /frontend @dependabot (#774)
- Bump @typescript-eslint/parser from 4.29.3 to 4.30.0 @dependabot (#773)
- Bump ws from 7.5.3 to 7.5.4 @dependabot (#770)
- Bump tar from 6.1.4 to 6.1.11 in /tray @dependabot (#779)
- Bump tar from 6.1.4 to 6.1.11 in /frontend @dependabot (#781)
- Bump @typescript-eslint/eslint-plugin from 4.29.3 to 4.30.0 @dependabot (#771)
- Bump tar from 6.1.4 to 6.1.11 @dependabot (#780)
- Bump @types/node from 16.7.9 to 16.7.10 in /frontend @dependabot (#785)
- Bump systeminformation from 5.8.6 to 5.8.7 @dependabot (#782)
- Bump nest-winston from 1.5.0 to 1.6.0 @dependabot (#784)
- Bump @types/node from 16.7.9 to 16.7.10 @dependabot (#783)
- Bump @typescript-eslint/parser from 4.30.0 to 4.31.0 @dependabot (#794)
- Bump @types/react from 17.0.19 to 17.0.20 @dependabot (#793)
- Bump axios from 0.21.2 to 0.21.4 @dependabot (#792)
- Bump docker/setup-buildx-action from 1.5.1 to 1.6.0 @dependabot (#791)
- Bump @mdi/js from 5.9.55 to 6.1.95 in /frontend @dependabot (#796)
- Bump @types/react from 17.0.19 to 17.0.20 in /frontend @dependabot (#797)
- Bump @typescript-eslint/eslint-plugin from 4.30.0 to 4.31.0 @dependabot (#795)
- Bump axios from 0.21.2 to 0.21.4 in /frontend @dependabot (#798)
- Bump @types/node from 16.7.10 to 16.7.13 @dependabot (#800)
- Bump creyD/prettier_action from 3.3 to 4.0 @dependabot (#799)
- Bump jest from 27.1.0 to 27.1.1 @dependabot (#801)
- Bump @types/node from 16.7.13 to 16.9.1 @dependabot (#806)
- Bump stefanzweifel/git-auto-commit-action from 4.11.0 to 4.12.0 @dependabot (#805)
- Bump ws from 7.5.4 to 7.5.5 @dependabot (#802)
- Bump prettier from 2.3.2 to 2.4.0 @dependabot (#803)
- Bump eslint-import-resolver-typescript from 2.4.0 to 2.5.0 @dependabot (#808)
- Bump jest from 27.1.1 to 27.2.0 @dependabot (#813)
- Bump @typescript-eslint/parser from 4.31.0 to 4.31.1 @dependabot (#814)
- Bump eslint-import-resolver-typescript from 2.4.0 to 2.5.0 in /frontend @dependabot (#816)
- Bump @typescript-eslint/eslint-plugin from 4.31.0 to 4.31.1 @dependabot (#810)
- Bump pkg from 5.3.1 to 5.3.2 @dependabot (#809)
- Bump typescript from 4.4.2 to 4.4.3 @dependabot (#811)
- Bump typescript from 4.4.2 to 4.4.3 in /frontend @dependabot (#815)
- Bump systeminformation from 5.8.7 to 5.8.9 @dependabot (#812)
- Bump @types/react from 17.0.20 to 17.0.21 @dependabot (#818)
- Bump systeminformation from 5.8.9 to 5.9.1 @dependabot (#819)
- Bump @types/react from 17.0.20 to 17.0.21 in /frontend @dependabot (#820)
- Bump systeminformation from 5.9.1 to 5.9.2 @dependabot (#822)
- Bump prettier from 2.4.0 to 2.4.1 @dependabot (#821)
- Bump @types/node from 16.9.1 to 16.9.2 @dependabot (#823)
- Bump systeminformation from 5.9.2 to 5.9.3 @dependabot (#824)
- Bump @types/node from 16.9.2 to 16.9.4 @dependabot (#830)
- Bump @typescript-eslint/parser from 4.31.1 to 4.31.2 @dependabot (#832)
- Bump @types/react from 17.0.21 to 17.0.22 in /frontend @dependabot (#834)
- Bump jest from 27.2.0 to 27.2.1 @dependabot (#829)
- Bump @typescript-eslint/eslint-plugin from 4.31.1 to 4.31.2 @dependabot (#833)
- Bump @types/react from 17.0.21 to 17.0.22 @dependabot (#831)
- Bump tmpl from 1.0.4 to 1.0.5 @dependabot (#835)
- Bump @types/react from 17.0.22 to 17.0.23 in /frontend @dependabot (#839)
- Bump @types/react from 17.0.22 to 17.0.23 @dependabot (#838)
- Bump @types/jest from 27.0.1 to 27.0.2 @dependabot (#837)
- Bump ts-loader from 9.2.5 to 9.2.6 @dependabot (#836)
- Bump @types/react from 17.0.23 to 17.0.24 @dependabot (#841)
- Bump jest-junit from 12.2.0 to 12.3.0 @dependabot (#843)
- Bump @types/node from 16.9.4 to 16.9.6 @dependabot (#842)
- Bump @types/react from 17.0.23 to 17.0.24 in /frontend @dependabot (#844)
- Bump systeminformation from 5.9.3 to 5.9.4 @dependabot (#845)
- Bump @nestjs/testing from 8.0.6 to 8.0.7 @dependabot (#846)
- Bump @nestjs/websockets from 8.0.6 to 8.0.7 @dependabot (#848)
- Bump @nestjs/common from 8.0.6 to 8.0.7 @dependabot (#849)
- Bump @nestjs/platform-ws from 8.0.6 to 8.0.7 @dependabot (#850)
- Bump @nestjs/platform-express from 8.0.6 to 8.0.7 @dependabot (#851)
- Bump @nestjs/core from 8.0.6 to 8.0.7 @dependabot (#847)
- Fix Windows Sensors @timmo001 (#852)

<hr /><em>This discussion was created from the release <a href='https://github.com/timmo001/system-bridge/releases/tag/v2.6.0'>v2.6.0</a>.</em>

**Comment by boardwarp** [comment]
I've just updated and noticed some strange behaviour under Windows 10. The web interface is timing out, but there aren't any errors in the log, I'm just seeing:
```
2021-09-25 13:08:28 info: System Bridge 2.6.0: D:\Program Files\System Bridge\system-bridge.exe - D:\Program Files\System Bridge - ["D:\\Program Files\\System Bridge\\system-bridge.exe","C:\\snapshot\\system-bridge\\dist\\index.js"] -  -  -  - 
2021-09-25 13:08:28 info: Process Manager - Starting api - ["D:\\Program Files\\System Bridge\\system-bridge.exe","dist/components/api/index.js"]
2021-09-25 13:08:29 info: Process Manager - Starting tray - ["D:\\Program Files\\System Bridge\\system-bridge-tray.exe"]
2021-09-25 13:08:32 info: Tray - 2.6.0: D:\Program Files\System Bridge\system-bridge-tray.exe - D:\Program Files\System Bridge - ["D:\\Program Files\\System Bridge\\system-bridge-tray.exe","C:\\snapshot\\system-bridge\\tray\\dist\\index.js"] -  -  - D:\Program Files\System Bridge -  - 
2021-09-25 13:08:32 info: Tray - Create Tray
2021-09-25 13:08:32 info: Tray - Started
```
If I exit System Bridge from the tray icon, it appears to close successfully. However, each time I re-launch, the closed instance(s) seem to reopen in addition to one new instance. e.g. closing and reopening the first time results in two System Bridge tray icons, the second time results in three icons, etc. There are some errors corresponding to these events:
```
2021-09-25 13:10:21 error: Process Manager - tray: Exited with code: 1
2021-09-25 13:10:21 info: Process Manager - tray: Restarting in 10 seconds..
```
[Full logs here](https://pastebin.com/vShQCWwr). I started with an in-place update from 2.5.0 to 2.6.0, and when these issues came up, I uninstalled, deleted the Program Files and App Data folders (it seemed like the NodeJS runtime kept running and prevented clean uninstallation), rebooted and installed again. This didn't seem to make any difference. The MQTT service still seems to work, but not the API endpoints (unsure if this is for the same reason the web interface isn't working, or because I'm missing the regenerated api key)

**Reply by spitz66** [reply]
Same behaviour here, can't access settings or view data anymore.

**Reply by timmo001** [reply]
Nothing looks out of order from the logs there from what I can see.

Can you open the path the app is installed to where you can see both executables and the additional files:

![image](https://user-images.githubusercontent.com/28114703/134777556-a2ee10ad-6348-448d-8c40-2d39f1b92949.png)

From there, open this folder in a terminal/powershell and manually run the app from there:

![image](https://user-images.githubusercontent.com/28114703/134777380-784733fa-2f09-4dc1-8409-eefa0ce1e181.png)

> If the app is being ran as admin, you should either disable this temporarily in the properties or run an elevated terminal/powershell window and navigate to the directory from there.

You should then see the log output show. This may give some additional information that is not handled by the logger module

**Reply by spitz66** [reply]
Still happens in 2.7.2 (and version shows as 2.7.1) as below:

_2021-09-26 13:39:23 info: System Bridge 2.7.1: C:\Users\spitz66\AppData\Local\System Bridge - C:\Users\spitz66\AppData\Local\System Bridge\system-bridge.exe - C:\Users\spitz66\AppData\Local\System Bridge - ["C:\\Users\\spitz66\\AppData\\Local\\System Bridge\\system-bridge.exe","C:\\snapshot\\system-bridge\\dist\\index.js"] -  -  -  - 
2021-09-26 13:39:23 info: Process Manager - Starting api - ["C:\\Users\\spitz66\\AppData\\Local\\System Bridge\\system-bridge.exe","dist/components/api/index.js"]
2021-09-26 13:39:23 info: Process Manager - Starting tray - ["C:\\Users\\spitz66\\AppData\\Local\\System Bridge\\system-bridge-tray.exe"]
2021-09-26 13:39:26 info: Tray - 2.7.1: C:\Users\spitz66\AppData\Local\System Bridge - C:\Users\spitz66\AppData\Local\System Bridge\system-bridge-tray.exe - C:\Users\spitz66\AppData\Local\System Bridge - ["C:\\Users\\spitz66\\AppData\\Local\\System Bridge\\system-bridge-tray.exe","C:\\snapshot\\system-bridge\\tray\\dist\\index.js"] -  -  -  - 
2021-09-26 13:39:26 info: Tray - Create Tray
2021-09-26 13:39:27 info: Tray - Started
2021-09-26 13:40:29 error: Process Manager - tray: Exited with code: 1
2021-09-26 13:40:29 info: Process Manager - tray: Restarting in 10 seconds.._

When reverting to 2.5.0 everything works again except for CPU load issue.

**Comment by boardwarp** [comment]
When I try running via CMD, it does give some error but closes the window a split second later so I haven't been able to capture what it says. The same behaviour is happening for me on 2.7.3, but I'm not sure if the error might be related to the process being re-launched automatically and conflicting with the cmd process
